### PR TITLE
Optimize code to remove useless objects

### PR DESCRIPTION
**Pull request by srchen1987**
Optimize code to remove useless objects

update before
public void addAttachments(Map<String, String> attachments) {
if (attachments == null) {
return;
}
if (this.attachments == null) {
this.attachments = new HashMap<String, String>();
}
this.attachments.putAll(attachments);
}

update after
public void addAttachments(Map<String, String> attachments) {
if (attachments == null) {
return;
}
if (this.attachments == null) {
this.attachments = attachments;
}else{
this.attachments.putAll(attachments);
}
}

### CODE REVIEW

File: dubbo-compatible/src/main/java/com/alibaba/dubbo/rpc/RpcInvocation.java [hunk 1]
```diff
@@ -160,15 +160,16 @@ public void setAttachmentIfAbsent(String key, String value) {
             attachments.put(key, value);
         }
     }
-
+    //Optimize code to remove useless objects
```
Comment (horizonzy) [inline dubbo-compatible/src/main/java/com/alibaba/dubbo/rpc/RpcInvocation.java:163]: The comment is unnecessary

File: dubbo-compatible/src/main/java/com/alibaba/dubbo/rpc/RpcInvocation.java [hunk 2]
```diff
@@ -160,15 +160,16 @@ public void setAttachmentIfAbsent(String key, String value) {
...
     public void addAttachments(Map<String, String> attachments) {
         if (attachments == null) {
             return;
         }
         if (this.attachments == null) {
-            this.attachments = new HashMap<String, String>();
+            this.attachments = attachments;
+        }else{
```
Comment (horizonzy) [inline dubbo-compatible/src/main/java/com/alibaba/dubbo/rpc/RpcInvocation.java:170]: indent problem.

**Comment by gonwan** [comment]
Hi there,

Happened to look through the release notes, but this merge introduced side effects. It assumes the argument `attachments` is not going to be modified outside the method, which is not always correct. Should revert this.

**Comment by horizonzy** [comment]
> Hi there,
> 
> Happened to look through the release notes, but this merge introduced side effects. It assumes the argument `attachments` is not going to be modified outside the method, which is not always correct. Should revert this.

yep, it's indeed risk.

**Pull request closed without merging**